### PR TITLE
Use buble transpiler and judiciously selected polyfills to reduce compiled Javascript bulk.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

To reduce bulk use a leaner transpiler that is a bit less flexible, and only use polyfills for things that we actually use in our codebase.

## TODO

- [ ] Have tests been written for the new code?
- [X] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

This is opened as a WIP/POC of switching transpiler to reduce bulk, would like some more testing to ensure it doesn't break anything.

## Results Summary

All results are for a full production, minified build.

| Module Name       | Main Bundle Before (kB) | Main Bundle After (kB) | Build Time Before (ms) | Build Time After (ms) |
| ----------------- | ----------------------- | ---------------------- | ---------------------- | --------------------- |
| audio             | 54.1                    | 36.4                   | 47284                  | 27598                 |
| video             | 463                     | 445                    | 47245                  | 27639                 |
| learn             | 118                     | 98.1                   | 46399                  | 39309                 |
| Management        | 110                     | 82.8                   |  47259                 | 29378                 |
| document          | 30.4                    | 12.7                   | 47250                  | 15998                 |
| setup_wizard      | 37.9                    | 12.6                   | 46407                  | 18043                 |
| default_frontend  | 1.11MB                  | 1.02MB                 | 47248                  | 39313                 |

## Before

```
Version: webpack 1.13.1
Child audio_mp3_render_module:
    Hash: 7093ce91b1f69c8418df
    Version: webpack 1.13.1
    Time: 47284ms
                                 Asset     Size  Chunks             Chunk Names
    audio_mp3_render_module-0.1.1c1.js  54.1 kB       0  [emitted]  audio_mp3_render_module
        + 100 hidden modules
Child video_mp4_render_module:
    Hash: bf31a042c90572189c6b
    Version: webpack 1.13.1
    Time: 47245ms
                                           Asset     Size  Chunks             Chunk Names
    VideoJS.eot?9499bb60a03b7e136a699cd87b159cc0  7.25 kB          [emitted]  
    VideoJS.eot?00103e881a36640a08f869ff6888f0fd   5.9 kB          [emitted]  
              video_mp4_render_module-0.1.1c1.js   463 kB       0  [emitted]  video_mp4_render_module
        + 105 hidden modules
    
    WARNING in kolibri/plugins/video_mp4_render/~/video.js/dist/video.js
    Critical dependencies:
    13:480-487 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
     @ kolibri/plugins/video_mp4_render/~/video.js/dist/video.js 13:480-487
Child management_module:
    Hash: 8b7c22057eb23ff14452
    Version: webpack 1.13.1
    Time: 47259ms
                           Asset    Size  Chunks             Chunk Names
    management_module-0.1.1c1.js  110 kB       0  [emitted]  management_module
        + 202 hidden modules
Child document_pdf_render_module:
    Hash: cafbddd602e0070646d7
    Version: webpack 1.13.1
    Time: 47250ms
                                    Asset     Size  Chunks             Chunk Names
    document_pdf_render_module-0.1.1c1.js  30.4 kB       0  [emitted]  document_pdf_render_module
        + 99 hidden modules
Child setup_wizard:
    Hash: 30dfe5a9a4952f68f224
    Version: webpack 1.13.1
    Time: 46407ms
                                            Asset     Size  Chunks             Chunk Names
    logo-min.png?eb2802030a74ce18b835e13478e6c62e  13.4 kB          [emitted]  
                          setup_wizard-0.1.1c1.js  37.9 kB       0  [emitted]  setup_wizard
        + 119 hidden modules
Child default_frontend:
    Hash: 46c98c84dad1527d5e7d
    Version: webpack 1.13.1
    Time: 47248ms
                                                   Asset     Size  Chunks             Chunk Names
    loading-spinner.gif?e850c29286b4eb51d1e5e00738b3fc04  98.8 kB          [emitted]  
       kolibri-logo.svg?e9efee91576fa4fabe5eb6a15b52ddd9  15.7 kB          [emitted]  
                             default_frontend-0.1.1c1.js  1.11 MB       0  [emitted]  default_frontend
                                          1.1-0.1.1c1.js  58.9 kB       1  [emitted]  
        + 624 hidden modules
Child learn_module:
    Hash: e0cac345df7b2c24a0e8
    Version: webpack 1.13.1
    Time: 46399ms
                      Asset    Size  Chunks             Chunk Names
    learn_module-0.1.1c1.js  118 kB       0  [emitted]  learn_module
        + 217 hidden modules
```

## After
```
Child video_mp4_render_module:
    Hash: bcfbea42f42ef2a3b85f
    Version: webpack 1.13.1
    Time: 27639ms
                                           Asset     Size  Chunks             Chunk Names
    VideoJS.eot?9499bb60a03b7e136a699cd87b159cc0  7.25 kB          [emitted]  
    VideoJS.eot?00103e881a36640a08f869ff6888f0fd   5.9 kB          [emitted]  
              video_mp4_render_module-0.1.1c1.js   445 kB       0  [emitted]  video_mp4_render_module
        + 19 hidden modules
    
    WARNING in kolibri/plugins/video_mp4_render/~/video.js/dist/video.js
    Critical dependencies:
    13:480-487 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
     @ kolibri/plugins/video_mp4_render/~/video.js/dist/video.js 13:480-487
Child audio_mp3_render_module:
    Hash: 2e5cd00e9c24cc16825a
    Version: webpack 1.13.1
    Time: 27598ms
                                 Asset     Size  Chunks             Chunk Names
    audio_mp3_render_module-0.1.1c1.js  36.4 kB       0  [emitted]  audio_mp3_render_module
        + 14 hidden modules
Child document_pdf_render_module:
    Hash: 96748857de9a792d24a3
    Version: webpack 1.13.1
    Time: 15998ms
                                    Asset     Size  Chunks             Chunk Names
    document_pdf_render_module-0.1.1c1.js  12.7 kB       0  [emitted]  document_pdf_render_module
        + 13 hidden modules
Child default_frontend:
    Hash: 1d77306adc9c149b774f
    Version: webpack 1.13.1
    Time: 39313ms
                                                   Asset     Size  Chunks             Chunk Names
    loading-spinner.gif?e850c29286b4eb51d1e5e00738b3fc04  98.8 kB          [emitted]  
       kolibri-logo.svg?e9efee91576fa4fabe5eb6a15b52ddd9  15.7 kB          [emitted]  
                             default_frontend-0.1.1c1.js  1.02 MB       0  [emitted]  default_frontend
                                          1.1-0.1.1c1.js  58.9 kB       1  [emitted]  
        + 306 hidden modules
Child learn_module:
    Hash: 4620b908ff0466379431
    Version: webpack 1.13.1
    Time: 39309ms
                      Asset     Size  Chunks             Chunk Names
    learn_module-0.1.1c1.js  98.1 kB       0  [emitted]  learn_module
        + 122 hidden modules
Child setup_wizard:
    Hash: 0c147fd9d1d678dda4bc
    Version: webpack 1.13.1
    Time: 18043ms
                                            Asset     Size  Chunks             Chunk Names
    logo-min.png?eb2802030a74ce18b835e13478e6c62e  13.4 kB          [emitted]  
                          setup_wizard-0.1.1c1.js  12.6 kB       0  [emitted]  setup_wizard
        + 17 hidden modules
Child management_module:
    Hash: 7c1238b3da5eb13202ea
    Version: webpack 1.13.1
    Time: 29378ms
                           Asset     Size  Chunks             Chunk Names
    management_module-0.1.1c1.js  82.8 kB       0  [emitted]  management_module
        + 89 hidden modules
```

### CODE REVIEW

**Comment by rtibbles** [comment]
Need to polyfill our tests too...

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
   "private": true,
   "dependencies": {
     "babel-runtime": "6.9.2",
+    "core-js": "^2.4.1",
```
Comment (indirectlylit) [inline package.json:32]: please remove the `^`s from this file

Comment (rtibbles) [reply package.json:32]: Will do on the other PR!

**Comment by indirectlylit** [comment]
superseded by #782

**Pull request closed without merging**